### PR TITLE
ci(.github): enable self hosted runners for AMD64 E2E tasks

### PR DESCRIPTION
**Pull request by jijiechen**
### Checklist prior to review

Set the runner dynamically to be used before invoking `_e2e.yaml` from `_test.yaml` to support switching runners based on whether a build is trusted or not.

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - fixes https://github.com/kumahq/kuma/issues/10585
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS 
  - Confirmed
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) 
  - Manually tested
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? 
  - No need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) 
- Added

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by jijiechen** [comment]
Here is a test run with self hosted runners enabled on e2e:
https://github.com/kumahq/kuma/actions/runs/9775105506

**Comment by jakubdyszkiewicz** [comment]
PR looks good to me, but I see that sample run fails :(

**Comment by jijiechen** [comment]
Yeah, some tasks failed at `make dev/tools`, I'll take a look.

**Review by jakubdyszkiewicz — APPROVED** [review]

File: .github/workflows/_test.yaml [hunk 1]
```diff
@@ -95,6 +96,48 @@ jobs:
...
+              ALL_RUNNERS=$(echo $ALL_RUNNERS | jq -r ".$ARCH=\"\"")
+            else
+              ALL_RUNNERS=$(echo $ALL_RUNNERS | jq -r ".$ARCH=.$ARCH[0]")
+            fi
+          done
+
+          echo "final runners: $ALL_RUNNERS"
+          echo "runners<<EOF" >> $GITHUB_OUTPUT
+          echo "$ALL_RUNNERS" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
```
Comment (lahabana) [inline .github/workflows/_test.yaml:140]: Isn't that just:

```shell
ALL_RUNNERS='{"amd64": "ubuntu-latest", "arm64": "ubuntu-latest-arm64"}'
if [[ "${{ evv.IS_TRUSTED_BUILD}}" == "true" ]]; then
  ALL_RUNNERS='{"amd64": "ubuntu-latest-kong", "arm64": "ubuntu-latest-arm64-kong"}'
fi

echo "final runners: $ALL_RUNNERS"
echo "runners<<EOF" >> $GITHUB_OUTPUT
echo "$ALL_RUNNERS" >> $GITHUB_OUTPUT
echo "EOF" >> $GITHUB_OUTPUT
```

?

Then in KM we can simply override `IS_TRUSTED_BUILD` to always be true.

**Pull request merged**